### PR TITLE
interfaces/mount: add dedicated mount entry type

### PR DESCRIPTION
**Pull request by zyga**
This patch adds a MountEntry structure that contains broken-down fields
present in an /etc/fstab record. They will soon replace the simple
string based snippets.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: interfaces/mount/entry.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+//     char *Mnt_type;     /* mount type (see Mntent.h) */
+//     char *Mnt_opts;     /* mount options (see Mntent.h) */
+//     int   Mnt_freq;     /* dump frequency in days */
+//     int   Mnt_passno;   /* pass number on parallel fsck */
+// };
+type Entry struct {
+	FsName  MntFsName
+	Dir     MntDir
+	FsType  MntFsType
+	Options MntOptions
```
Comment (niemeyer) [inline interfaces/mount/entry.go]: This feels over-typed.  Can we just have string/[]string on those, and a single `Entry.String` method?

Also, this feels neither aligned with the original names, nor nice to read. Going one way or the other seems fine, but the middle is a bit awkward.

For example, this reads nicely:

```
type Entry struct {
        Name    string
        Dir     string
        Type    string
        Options []string

        DumpFrequency   int
        CheckPassNumber int
}
```

While this reads not as nicely, but is aligned with the original code:
```
type Entry struct {
        Fsname string
        Dir    string
        Type   string
        Opts   []string
        Freq   int
        Passno int
}
```

Either seem fine and better than different and mangled.

Comment (zyga) [inline interfaces/mount/entry.go]: You are right. I'll go ahead with non-legacy names (go did this a while ago and I think it was regarded as a mistake). As for the over-typing I think it is fine to use strings as long as the String() function does the special casing of empty values (which is easy).

Thanks for the feedback. I'll tweak this and push again.

File: interfaces/mount/entry.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	return whitespaceReplacer.Replace(s)
+}
+
+var whitespaceReplacer = strings.NewReplacer(
+	" ", `\040`, "\t", `\011`, "\n", `\012`, "\\", `\134`)
+
+func (e Entry) String() string {
+	// Name represents name of the device in a mount entry.
+	name := "none"
+	if len(e.Name) != 0 {
```
Comment (niemeyer) [inline interfaces/mount/entry.go]: `if e.Name != "" {` is a better spelling for these.

Comment (zyga) [inline interfaces/mount/entry.go]: Done

**Pull request merged**